src/lib.rs

```diff
@@ -17,6 +17,7 @@ mod test;
 pub struct Heap {
     bottom: usize,
     size: usize,
+    max_size: usize,
```

Comment:
Thanks for the review! Yes, that makes sense, we could leave it to the user to manage the complexity of the heap size. Also, one can then choose between allocating small heap size or going for on demand mapping. I will post the revised version in the coming days. Cheers